salt/_pillar/metalk8s_nodes.py

```diff
@@ -92,9 +96,40 @@ def ext_pillar(minion_id, pillar, kubeconfig):
             for node in node_list.items
         )
 
+        custom_object = []
+
+        try:
+            object_response = customObjectsApi.list_cluster_custom_object(
+                group="storage.metalk8s.scality.com", version="v1alpha1",
+                plural="volumes"
+            )
+            if not object_response:
+                custom_object = __utils__['pillar_utils.errors_to_dict']([
+                    'Unable to retrieve node volume list'
+                ])
+            for node in node_list.items:
+                for k, v in object_response.items():
+                    if k == "items":
+                        for items in v:
+                            volume_type = list(items['spec'].keys())[0]
+                            result_vol_object = {
+                             'volume_name': items['metadata']['name'],
+                             'volume_uid': items['metadata']['uid'],
+                             'volume_type': volume_type,
+                             'volume_capacity': items['spec'][
```

Comment:
This won't fly. You can't return the same (structure) of data for every type of volume.

Reply:
@NicolasT Thanks for picking this out. I am going to fix this.